File: java/PointersExample.java
//Java doesn't support pointers, but reference variables can be used similarly.

public class PointersExample {
    public static void main(String[] args) {
        int[] arr = {1, 2, 3};
        modifyArray(arr);
        System.out.println(arr[0]);
    }

    public static void modifyArray(int[] arr) {
        arr[0] = 100;
    }
}

File: csharp/PointersExample.cs
//C# supports pointers, but they require unsafe code blocks.

using System;

class PointersExample {
    unsafe static void Main(string[] args) {
        int a = 10;
        int* p = &a;
        modifyPointer(p);
        Console.WriteLine(a);
    }

    unsafe static void modifyPointer(int* p) {
        *p = 100;
    }
}
